google-drive/src/main/java/com/ivy/googledrive/backup/DriveBackupRepository.kt

```diff
@@ -0,0 +1,21 @@
+package com.ivy.googledrive.backup
+
+import androidx.work.WorkManager
+import com.ivy.googledrive.google_drive.DriveMimeType
+import java.io.File
+import kotlin.jvm.Throws
+
+interface DriveBackupRepository {
+    /**
+     * @param n is The capacity of the Backup folder,
+     * if the capacity is full then the oldest file will get deleted and get replaced by the new file.
+     * */
+    @Throws(Exception::class)
```

Comment:
That's very bad! Throwing exceptions leads to unexpected crash. Please use Arrow's `Either` like we did in https://github.com/Ivy-Apps/ivy-wallet/blob/deprecated-ivywallet-rewrite-2223/drive/google-drive/src/main/java/com/ivy/drive/google_drive/GoogleDriveServiceImpl.kt

Reply:
I catch the exceptions using try and catch

Reply:
Hey @mohammednawas8 that works but the problem is that Exceptions are prone to errors and runtime crashes. We work with typed errors:
https://arrow-kt.io/learn/typed-errors/working-with-typed-errors/

**TL;DR;**

- If a function is expected to fail, e.g. Network request (it happens often) we model its return type explicitly with `Either`
- We throw Exceptions only for exceptional cases - situations which we don't expect to happen. For example, save to DB failing because the device is out of storage or some IO error.

Does that make sense?

Reply:
This can provide more context:
https://www.linkedin.com/posts/iliyan-germanov-3963b5b9_working-with-typed-errors-arrow-activity-7115666238504103937-d5df?utm_source=share&utm_medium=member_android